commit 39adc895b1bd3455de3cea8f77a0aa6252116a04
Author: Andrew Beekhof <andrew@beekhof.net>
Date:   Mon Jan 12 13:53:22 2015 +1100

    Leave SBD_WATCHDOG_TIMEOUT in sectods

diff --git a/src/sbd-inquisitor.c b/src/sbd-inquisitor.c
index 52ca9ff..918dc35 100644
--- a/src/sbd-inquisitor.c
+++ b/src/sbd-inquisitor.c
@@ -767,7 +767,7 @@ int main(int argc, char **argv, char **envp)
 
         value = getenv("SBD_WATCHDOG_TIMEOUT");
         if(value) {
-            timeout_watchdog = crm_get_msec(value);
+            timeout_watchdog = crm_get_msec(value) / 1000;
         }
 
         value = getenv("SBD_PIDFILE");
